src/main/java/com/exasol/sql/expression/function/AbstractFunction.java

```diff
@@ -3,32 +3,42 @@
 import java.util.List;
 
 import com.exasol.sql.expression.ValueExpression;
-import com.exasol.util.AbstractTreeNode;
+import com.exasol.sql.expression.ValueExpressionVisitor;
 
 /**
  * This is a base class for functions.
  */
-public abstract class AbstractFunction extends AbstractTreeNode implements Function {
+public abstract class AbstractFunction implements Function {
     protected String functionName;
     protected List<ValueExpression> valueExpressions;
```

Comment:
From the perspective of the surrounding class, those are function parameters or simply parameters. Value expression is a type, not an attribute. Please rename.

I know it was in the original code, but that sometimes happens during reviews. :-)